### PR TITLE
#613 - Add the possibility to change date formatting

### PR DESCRIPTION
**Pull request by kisiseldwarf**

### CODE REVIEW

File: totalRP3/Modules/Register/Main/RegisterList.lua [hunk 1]
```diff
@@ -1083,3 +1082,7 @@ TRP3_API.events.listenToEvent(TRP3_API.events.WORKFLOW_ON_LOADED, function()
 		});
 	end
 end);
+
+function getDateSettings()
```
Comment (Meorawr) [inline totalRP3/Modules/Register/Main/RegisterList.lua:1086]: This needs to not be a global function; would recommend renaming it and all usages to `TRP3_API.util.getPreferredDateFormat` for now as that'll likely be what's wanted by Extended.

File: totalRP3/Core/Enums.lua [hunk 2]
```diff
@@ -44,3 +44,13 @@ AddOn_TotalRP3.Enums.UNIT_TYPE = {
 	MOUNT = "MOUNT", -- Not a real unit, but used for companion profiles.
 	NPC = "NPC",
 };
+
+-- Configuration --
+AddOn_TotalRP3.Enums.GENERAL_SETTINGS_CONFIG_KEYS = {
+	DATE_FORMAT = "date_format"
```
Comment (Meorawr) [inline totalRP3/Core/Enums.lua:50]: For now I'm tempted to suggest just hardcoding the "date_format" config key in the areas it's being used; longer-term I want to rework the config key nonsense and would rather not have to _also_ factor in this enum,

File: totalRP3/Core/Configuration.lua [hunk 3]
```diff
@@ -378,12 +378,17 @@ TRP3_API.events.listenToEvent(TRP3_API.events.WORKFLOW_ON_LOAD, function()
...
+	local date_format_settings_key = AddOn_TotalRP3.Enums.GENERAL_SETTINGS_CONFIG_KEYS.DATE_FORMAT;
+	local date_option_enum = AddOn_TotalRP3.Enums.DATE_OPTIONS;
+
 	registerConfigKey("heavy_profile_alert", true);
 	registerConfigKey("new_version_alert", true);
 	registerConfigKey("ui_sounds", true);
 	registerConfigKey("ui_animations", true);
 	registerConfigKey("default_color_picker", false);
 	registerConfigKey("increase_color_contrast", false);
+	registerConfigKey(date_format_settings_key, date_option_enum.D_M_Y);
```
Comment (Meorawr) [inline totalRP3/Core/Configuration.lua:391]: I'll need to think some more on the default here. That currently we're using D/M/Y makes this at least consistent, but I feel as if this should actually be "intelligent" and use a proper default based off client locale - but that might confuse US players if we now swapped D/M/Y to M/D/Y for them.

File: totalRP3/Locales/Locale.lua [hunk 4]
```diff
@@ -624,6 +624,7 @@ Possible status:
 	CO_LOCATION_DISABLE_CLASSIC_PVP_TT = "You will not respond to location requests from other players when you are flagged for PvP.\n\nThis option is particularly useful to avoid abuses of the location system to track you.",
 	CO_SANITIZER = "Sanitize incoming profiles",
 	CO_SANITIZER_TT = "Remove escaped sequences in tooltip fields from incoming profiles when TRP doesn't allow it (color, images ...).",
+	CO_DATE_FORMAT = "Date Format",
```
Comment (Meorawr) [inline totalRP3/Locales/Locale.lua:627]: There's a header section further down the Locale script for adding new localization strings - it should go underneath that section.

**Review by Meorawr — CHANGES_REQUESTED** [review]

File: totalRP3/Core/Enums.lua (hunk 2 above)
Comment (kisiseldwarf) [reply totalRP3/Core/Enums.lua:50]: Oof, yeah it hurts to do that but I understand that my changes would add more areas to look over for your refacto !

File: totalRP3/Core/Configuration.lua (hunk 3 above)
Comment (kisiseldwarf) [reply totalRP3/Core/Configuration.lua:391]: I agree that making it smart would only confuse players on why their settings has changed without their approval

**Review by Meorawr — DISMISSED** [review]
Minus the random whitespace stuff Luacheck complained about it looks fine - we'll think about the default format some more over the weekend and tweak it if necessary.

**Comment by kisiseldwarf** [comment]
Oops. didn't know it would dismiss your review (First time I do a PR on a collaborative project :sweat_smile:

**Review by Solanya — APPROVED** [review]

**Comment by Solanya** [comment]
Thanks, will be sure to credit your help in the changelog 👍

**Comment by kisiseldwarf** [comment]
No problem !
I would love to continue to contribute to this project, if you need anything done let me know

**Pull request merged**